### ISSUE TITLE
Recommend Contract.check methods use **kwargs

### ISSUE THREAD

**Issue by seddonym** [seddonym/import-linter#139]
It would improve backwards compatibility for Contract.check methods to implement their check methods like so, in case we want to add arguments:

```
check(graph: ImportGraph, verbose: bool, **kwargs) -> ContractCheck:
```

Prompted by https://github.com/seddonym/import-linter/issues/138.

**Comment by mwgamble** [comment]
Perhaps a better approach would be to create a `CheckOptions` class, and move the `verbose` parameter in there. Then you can add additional (optional) parameters to that class as desired, and plugin implementers don't have to adjust their code at all just because a new parameter is added.

**Comment by seddonym** [comment]
Good point. It would result in a breaking change though so I think the change would only be appropriate for a major version.
